PixyViSy.cpp

```diff
@@ -1,91 +1,116 @@
 #include <PixyViSy.h>
 
-PixyViSy::PixyViSy(uint8_t sig, uint16_t calibration_dis_pix, uint8_t _goal_height)
+const uint16_t min_block_area = 100;
+const uint16_t ball_size = 12;
+const uint16_t pixy_screen_half = (PIXY_MAX_X +  1) / 2;
+
+inline uint16_t block_size(Block &block)
+{
+    return block.height * block.width;
+}
+
+PixyViSy::PixyViSy(uint8_t sig, uint16_t calib_dis_pix, uint8_t _goal_height)
 {
-  pixy.init();
-  signature = sig;
-  calib_const = calibration_dis_pix;
-  left_pixels = 0, right_pixels = 0, blocks_count = 0;
-  goal_height = _goal_height;
-  action = 'N';
+    pixy.init();
+    signature = sig;
+    calib_const = calib_dis_pix;
+    left_pixels = 0, right_pixels = 0, blocks_count = 0;
+    goal_height = _goal_height;
+    action = 'N';
 }
+
 void PixyViSy::update(void)
 {
-  blocks_count = 0;
-  uint8_t all_blocks_count = pixy.getBlocks();
-  left_pixels = 0;
-  right_pixels = 0;
-  action = 'N'; // as "Nothing"
-  distance = ~0;
-  goal_pix_height = 0;
-  uint8_t max[3] = { all_blocks_count, all_blocks_count, all_blocks_count };
+    // set default values
+    blocks_count = 0;
+    left_pixels = 0;
+    right_pixels = 0;
+    action = 'N'; // as "Nothing"
+    distance = ~0;
+    goal_pix_height = 0;
+    uint8_t all_blocks_count = pixy.getBlocks();
+    uint8_t max[3] = { all_blocks_count, all_blocks_count, all_blocks_count };
 
-  if (all_blocks_count) {
-    /* finding three max blocks */
-    for (uint8_t i = 0; i < all_blocks_count; i++) {
-      if (pixy.blocks[i].signature == signature && BLOCK_SIZE(pixy.blocks[i]) >= MIN_BLOCK_AREA) {
-        if (max[2] == all_blocks_count || BLOCK_SIZE(pixy.blocks[i]) > BLOCK_SIZE(pixy.blocks[max[2]])) {
-          if (max[1] == all_blocks_count || BLOCK_SIZE(pixy.blocks[i]) > BLOCK_SIZE(pixy.blocks[max[1]])) {
-            if (max[0] == all_blocks_count || BLOCK_SIZE(pixy.blocks[i]) > BLOCK_SIZE(pixy.blocks[max[0]])) {
-              max[2] = max[1];
-              max[1] = max[0];
-              max[0] = i;
-            } else {
-              max[2] = max[1];
-              max[0] = i;
+    if (all_blocks_count) {
+        // finding three max blocks
+        for (uint8_t i = 0; i < all_blocks_count; i++) {
+            if (pixy.blocks[i].signature == signature
+            && block_size(pixy.blocks[i]) >= min_block_area) {
```

Comment:
@vargac To be honest, I am not sure if this is an actual improvement, in terms of readability.